jsonpb/jsonpb.go

```diff
@@ -637,15 +637,18 @@ func UnmarshalString(str string, pb proto.Message) error {
 func (u *Unmarshaler) unmarshalValue(target reflect.Value, inputValue json.RawMessage, prop *proto.Properties) error {
 	targetType := target.Type()
 
-	// Allocate memory for pointer fields.
-	if targetType.Kind() == reflect.Ptr {
+	switch targetType.Kind() {
+	case reflect.Map, reflect.Ptr, reflect.Slice:
```

Comment:
Done, and added tests. Thanks for making me do that; this implementation turned out not to be correct.